src/withJWTProps.js

```diff
@@ -0,0 +1,57 @@
+import React, { Component } from 'react'
+import PropTypes from 'prop-types'
+import wrapDisplayName from 'recompose/wrapDisplayName'
+import jwtDecode from 'jwt-decode'
+
+export default (propName, propsMapping) => Target => {
+  class WithJWTProps extends Component {
+    constructor(props, context) {
+      super(props, context)
+
+      this.state = this.buildState(props[propName])
+    }
+
+    componentWillReceiveProps(nextProps) {
+      if (nextProps[propName] !== this.props[propName]) {
+        this.setState(this.buildState(nextProps[propName]))
+      }
+    }
+
+    buildState(token) {
+      try {
+        const decodedToken = jwtDecode(token)
```

Comment:
@Nevon `jwt-decode` parses the token without validating its signature. So, as long as the token payload is well formed, it will be able to parse it. Does it make sense to have such validation on the client side?